### PR TITLE
TEM-149: node resource quotas

### PR DESCRIPTION
**Pull request by bobheadxi**
## :construction_worker: Purpose

Restrict disk, memory, and cpu

## :rocket: Changes

Various refactors to accommodate this feature. Will do a detailed writeup later, have class now 😛 

## :warning: Breaking Changes

Label refactors - all nodes will need to be removed and recreated

Depends on https://github.com/RTradeLtd/database/pull/15

### CODE REVIEW

**Review by bonedaddy — COMMENTED** [review]
@bobheadxi Unless I'm missing something, it doesn't look like the nodes are being created with a swarm key? If so this would end up being a public node

File: ipfs/ipfs_utils.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	return scanner.Err()
+}
+
+func (c *client) initNodeFS(n *NodeInfo, opts NodeOpts) error {
+	// set up directories
+	os.MkdirAll(c.getDataDir(n.NetworkID), c.fileMode)
+
+	// write swarm.key to mount point, otherwise check if a swarm key exists
+	keyPath := c.getDataDir(n.NetworkID) + "/swarm.key"
+	if opts.SwarmKey != nil {
```
Comment (bonedaddy) [inline ipfs/ipfs_utils.go:52]: Nevermind i was missing something, looks like the key is writen here.

**Comment by bonedaddy** [comment]
@bobheadxi  can you review these [logs](https://gist.github.com/postables/00dfa3fa5b7f7bc8404edbbc4fff509e)? Some errors/warnings occurred that could could just be due to the kernel I'm using.

Couple things, an issue occurred during container creation [see here](https://gist.github.com/postables/00dfa3fa5b7f7bc8404edbbc4fff509e#file-test-log-L33) however the issue check during startup didn't look like it recognized the error, [see here](https://gist.github.com/postables/00dfa3fa5b7f7bc8404edbbc4fff509e#file-test-log-L42)

Not sure if these are of concern or not.

File: ipfs/ipfs_utils.go (hunk 1 above)
Comment (bobheadxi) [reply ipfs/ipfs_utils.go:52]: ah my bad, the naming isn't very clear - i'll improve the docs and function name

**Review by bonedaddy — APPROVED** [review]
Excellent :fire:  :100: :tada:

**Pull request merged**